test/widgets/app_bar_test.dart

```diff
@@ -35,4 +37,124 @@ void main() {
     check(tester.getRect(find.byType(ZulipAppBar))).equals(rectBefore);
     check(finder.evaluate()).single;
   });
+
+  group("buildTitle's willCenterTitle agrees with Material AppBar", () {
+    /// Build an [AppBar]; inspect and return whether it decided to center.
+    Future<bool> material(WidgetTester tester, {
+      required bool? paramValue,
+      required bool? themeValue,
+      required List<Widget>? actions,
+    }) async {
+      testBinding.reset();
+
+      final themeData = ThemeData(appBarTheme: AppBarTheme(centerTitle: themeValue));
+      final widget = TestZulipApp(
+        child: Theme(data: themeData,
+          child: AppBar(
+            centerTitle: paramValue,
+            actions: actions,
+            title: const Text('a'))));
+
+      await tester.pumpWidget(widget);
+      await tester.pump();
+
+      // test assumes LTR text direction
+      check(tester.platformDispatcher.locale).equals(const Locale('en', 'US'));
+      assert(actions == null || actions.isNotEmpty);
+      final titleAreaRightEdgeOffset = actions == null
+        ? (tester.view.physicalSize / tester.view.devicePixelRatio).width
+        : tester.getTopLeft(find.byWidget(actions.first)).dx;
+      final titlePosition = tester.getCenter(find.text('a')).dx;
+      final isCentered = titlePosition > ((1 / 3) * titleAreaRightEdgeOffset);
+      check(titlePosition).isLessThan((2 / 3) * titleAreaRightEdgeOffset);
```

Comment:
How about:
```suggestion
      final titlePosition = tester.getTopLeft(find.text('a')).dx;
      final isCentered = titlePosition > ((1 / 3) * titleAreaRightEdgeOffset);
      check(titlePosition).isLessThan((2 / 3) * titleAreaRightEdgeOffset);
```

Otherwise this test implicitly relies on the title having its size shrink to its contents, rather than filling its parent's size.